docs/features/map-tiles-custom-areas/high-level-design.md

```diff
@@ -0,0 +1,37 @@
+# Role base access control - High-level design
+
+This document aims at providing an overview of the architecture of the tiles to fullfil the [requirements document](./requirements.md) for the Marxan platform.
+
+## Preview custom planning area tiler
+
+In order to generate the vector tiles for the custom planning area, we need to access to the `planning_areas` table in the marxan geoprocessing database. For visualization purposes, we only need to access to the `the_geom` column.
+
+The tiler system will follow a similar implementation as the other tiler systems within this project. In the geoprocessing service a custom query will be used to access the `the_geom` column of the `planning_areas` table and to generate the tiles.
+The data upload jobs should provide back the id of the planning area id. this information should be use to query the `the_geom` column of the `planning_areas` table. This will then be served to the user as a proxy of the mvt tile in the api service.
```

Comment:
ah ok, this is for the preview, I realised this only when I got to the _Project planning area tiler and Project planning grid tiler_ section below 😬 

Let's maybe clarify what the preview tile endpoints are for, in the intro of the doc? for example,

> The tiles needed to render all the kinds of areas reviewed in this document will be of two kinds:
> - _preview_: these are used when a user is configuring planning area and grid for a project, before the project is actually created in the system (hence, we won't have a `project_id` available at this stage, nor will any spatial data have been linked to a project yet, even though it will have been stored in the geodb
> - _for scenario_: these are used when project-level planning area and grid data is shown while creating or editing a scenario; at this stage the relevant spatial data will have been linked to the scenario's parent project

(the _for scenario_ label above sucks, but that's what I could think of... if you like the suggestion to explain in more detail the differences and want to keep something like my suggestion above, feel free to pick more meaningful names, of course 😅)